graphics/gscan2pdf/Portfile

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+
+PortGroup           perl5 1.0
+
+name                gscan2pdf
+version             2.12.4
+revision            0
+
+categories          graphics
+platforms           darwin
+maintainers         nomaintainer
+license             GPL-3
+description         A GUI to produce PDFs or DjVus from scanned documents
+long_description    ${description}
+
+homepage            http://gscan2pdf.sourceforge.net
+master_sites        sourceforge
+use_xz              yes
+
+checksums           rmd160  d800f0f54214763dacf0847f5eece92dcd076b65 \
+                    sha256  52b06dd1091293b20fe26658168c4534943542671be7709e9af9587ecc6367fb \
+                    size    503792
+
+depends_lib         port:ImageMagick \
+                    port:librsvg \
+                    port:p${perl5.major}-config-general \
+                    port:p${perl5.major}-date-calc \
+                    port:p${perl5.major}-filesys-df \
+                    port:p${perl5.major}-goocanvas2 \
+                    port:p${perl5.major}-gtk3 \
+                    port:p${perl5.major}-gtk3-imageview \
+                    port:p${perl5.major}-gtk3-simplelist \
+                    port:p${perl5.major}-image-sane \
+                    port:p${perl5.major}-ipc-system-simple \
+                    port:p${perl5.major}-list-moreutils \
+                    port:p${perl5.major}-locale-gettext \
+                    port:p${perl5.major}-log-log4perl \
+                    port:p${perl5.major}-pdf-builder \
+                    port:p${perl5.major}-perlmagick \
+                    port:p${perl5.major}-proc-processtable \
+                    port:p${perl5.major}-readonly \
+                    port:p${perl5.major}-set-intspan \
+                    port:p${perl5.major}-sub-override \
+                    port:p${perl5.major}-try-tiny \
+                    port:sane-backends \
+                    port:tiff \
+
+
+configure.cmd       ${perl5.bin}
+configure.pre_args  Makefile.PL --prefix=${prefix}
+
+# Optional dependencies
+#depends_lib-append  port:sane-frontends \
+#                    port:djvulibre \
+#                    port:tesseract \
+#                    port:gocr \
+#                    port:cuneiform
+#                    port:pdftk \
+#                    port:unpaper \
+#                    port:xdg-utils
```

Comment:
are these optional dependencies used / linked to in case the user has them installed (assuming they're available in MacPorts)? If so, that's not what we want to get reproducible builds. In that case you'll either want to make sure the dependencies are always installed or never used even if they are present.

Reply:
According to gscan2pdf, they are optional. They are not mandatory to build, neither to work but some features of the SW will be missing.

- pdftk is used for pdf encryption/decryption. **Pulls the full java (for pdftk-java package)**
- djvulibre to save as djvu
- xdg-util Required for Email as PDF.
- sane-frontends is only needed for scanadf frontend.
- For unpaper it will warn if the program is missing, and functionality is greyed out. **Pulls ~20 dependencies**
- For tesseract/gocr/cuneiform it will detect those installed and propose to choose one of them (if one wants to use OCR) If no OCR is present, the functionality is greyed out. The problem is see here is that tesseract for example **is a very large dependency** and we don't necessarily want it. Moreover, there is also gocr and cuneiform which do the same job. So I'm not sure it should be a strong dependency for the software.

I don't really now how to handle this. Would that be ok?
- put `xdg-utils`, `djvulibre` & `sane-frontends` as mandatory dependency (they don't look that big)
- **and** add these variants? `+tesseract +gocr +cuneiform +pdtk +unpaper`

Reply:
it all depends; if the program doesn't link to any of the other packages but only uses them if available at runtime then it doesn't matter and you can just leave them out. If not, the you'll do something about it as I said above. So you can check whether the resulting binary links to anything (using `otool -L <filename>` when you have some of the packages installed.

Reply:
BTW gscan2pdf is in perl, so no binary.

Reply:
I added a note.